Hyper/Meshes/Vertex.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Runtime.InteropServices;
+
+namespace Hyper.Meshes;
+
+[StructLayout(LayoutKind.Explicit)]
+internal struct Vertex
```

Comment:
We can rename it to Vertex3d to be consistent

Reply:
how about changing Vertex2d to HUDVertex? Also I think we might want to move HUD to another projectsince it doesnt interact with anything in this one. But let's leave that for a later PR.

Reply:
Then both could be called Vertex